src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java

```diff
@@ -111,6 +111,90 @@ private static Schema inferSchema(final String fieldPath, final BsonValue bsonVa
     }
   }
 
+  private static Schema combine(final Schema firstSchema, final Schema secondSchema) {
+    if (isSentinelValueSet(firstSchema)) {
```

Comment:
The initial value is always a sentinel because: 

`.reduce(SENTINEL_STRING_TYPE, BsonDocumentToSchema::combine)`

So we have to add a check for the initial value being the sentinel.

The rest falls through the logic.

Reply:
Ah, I was confusing this with the sentinel-checking logic in the `combineFieldSchema` method.